### PR TITLE
Follow-up to set-support work

### PR DESCRIPTION
**Pull request by kevmoo**
Related to https://github.com/dart-lang/json_serializable/issues/54
Follow-up on https://github.com/dart-lang/json_serializable/pull/302

### CODE REVIEW

File: json_serializable/lib/src/type_helpers/iterable_helper.dart [hunk 1]
```diff
@@ -70,8 +70,10 @@ class IterableHelper extends TypeHelper {
 
     var itemSubVal = context.deserialize(iterableGenericType, closureArg);
 
-    // If `itemSubVal` is the same, then we don't need to do anything fancy
-    if (closureArg == itemSubVal) {
+    // If `itemSubVal` is the same and it's not a Set, then we don't need to do
+    // anything fancy
+    if (closureArg == itemSubVal &&
```
Comment (kevmoo) [inline json_serializable/lib/src/type_helpers/iterable_helper.dart]: @gagoman – this was the one case that was missing. Tough to find unless you do the full set of cases in the `kitchen_sink` test

File: json_serializable/test/kitchen_sink/kitchen_sink.dart [hunk 2]
```diff
@@ -12,6 +12,7 @@ import 'strict_keys_object.dart';
 part 'kitchen_sink.g.dart';
 
 List<T> _defaultList<T>() => null;
```
Comment (jakemac53) [inline json_serializable/test/kitchen_sink/kitchen_sink.dart]: Can we get a comment here that these get overwritten to not return null in some cases :D

Comment (jakemac53) [reply json_serializable/test/kitchen_sink/kitchen_sink.dart]: Alternatively could you use like a zone variable or something to make this a bit less hacky?

Comment (kevmoo) [reply json_serializable/test/kitchen_sink/kitchen_sink.dart]: I'll add a comment. Let's keep the runtime clean.

**Pull request merged**